packages/flutter/test/material/date_picker_test.dart

```diff
@@ -721,6 +724,14 @@ void main() {
       });
     });
 
+    testWidgets('InputFormatters are used', (WidgetTester tester) async {
+      inputFormatters = <TextInputFormatter>[FilteringTextInputFormatter.allow(RegExp('[a-zA-Z]'))];
+      await prepareDatePicker(tester, (Future<DateTime?> date) async {
+        final TextField field = textField(tester);
+        expect(field.inputFormatters?.length, 1);
```

Comment:
This should either also assert that when inputFormatters is null the length is zero or (better) assert that the input formatters match the list of what was provided.